freespace2/freespace.cpp

```diff
@@ -1376,11 +1389,15 @@ bool game_start_mission()
 
 		return false;
 	}
-	load_mission_load = (uint) (time(nullptr) - load_mission_load);
 
-	// free up memory from parsing the mission
-	extern void stop_parse();
-	stop_parse();
+	// Since we just freed up memory, now is probably a good time to run the Lua garbage collector.
+	// This is also after the preliminary checks are complete but before things start getting paged in.
+	auto L = Script_system.GetLuaSession();
+	if (L != nullptr)
+	{
+		game_busy(NOX("** running Lua garbage collector **"));
```

Comment:
A more player-friendly message like
```
"** cleaning up Lua objects **"
```
might be nice, but then again, this message appears only in debug builds, so not that important.

Reply:
I like that; will do